File: app/src/main/java/ca/mitenko/evn/ui/dest_map/DestMapView.java
package ca.mitenko.evn.ui.dest_map;


import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.LatLngBounds;

import java.util.ArrayList;

import ca.mitenko.evn.model.Destination;
import ca.mitenko.evn.ui.common.RootView;

/**
 * Created by mitenko on 2017-04-23.
 */

public interface DestMapView extends RootView {
    /**
     * Sets the destination results
     * @param destinations
     */
    void setDestinations(ArrayList<Destination> destinations);

    /**
     * Sets the destination results
     * @param destination
     */
    void setSelectedItem(Destination destination);

    /**
     * Sets the map boundaries
     * @param mapBounds
     */
    void setMapBounds(LatLngBounds mapBounds, boolean animate);

    /**
     * Shows the Progress Bar
     */
    void showProgressBar();

    /**
     * Hides the Progress Bar
     */
    void hideProgressBar();
    /**
     * Shows the Update button
     */
    void showUpdateButton();

    /**
     * Hides the Update button
     */
    void hideUpdateButton();

    /**
     * Tells the map to recluster
     */
    void recluster();

    /**
     * Sets the user location
     */
    void setUserLocation(LatLng userLocation);
}

File: app/src/main/java/ca/mitenko/evn/ui/dest_map/DestMapView.kt
package ca.mitenko.evn.ui.dest_map


import ca.mitenko.evn.model.Destination
import ca.mitenko.evn.ui.common.RootView
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.LatLngBounds
import java.util.*

/**
 * Created by mitenko on 2017-04-23.
 */

interface DestMapView : RootView {
    /**
     * Displays an error
     */
    fun showError(message: String)

    /**
     * Sets the destination results
     * @param destinations
     */
    fun setDestinations(destinations: ArrayList<Destination>)

    /**
     * Sets the destination results
     * @param destination
     */
    fun setSelectedItem(destination: Destination?)

    /**
     * Sets the map boundaries
     * @param mapBounds
     */
    fun setMapBounds(mapBounds: LatLngBounds, animate: Boolean)

    /**
     * Shows the Progress Bar
     */
    fun showProgressBar()

    /**
     * Hides the Progress Bar
     */
    fun hideProgressBar()

    /**
     * Shows the Update button
     */
    fun showUpdateButton()

    /**
     * Hides the Update button
     */
    fun hideUpdateButton()

    /**
     * Tells the map to recluster
     */
    fun recluster()

    /**
     * Sets the user location
     */
    fun setUserLocation(userLocation: LatLng)
}
